### ISSUE TITLE
Wrong datatype for management.listener.https.verify?

### ISSUE THREAD

**Issue by RicardoHPeng** [emqx/emqx-management#29]
When I uncomment the https listener and start emqx, the mgmt plugins doesn't work. The error msg is about the verify option.
``
2019-01-21 14:53:28.781 [error] Failed to start Ranch listener 'https:management' in ranch_ssl:listen([{next_protocols_advertised,[<<"h2">>,<<"http/1.1">>]},{alpn_preferred_protocols,[<<"h2">>,<<"http/1.1">>]},{port,8081},{backlog,512},{send_timeout,15000},{send_timeout_close,true},{nodelay,true},{keyfile,"etc/certs/client.key"},{certfile,"etc/certs/client.crt"},{cacertfile,"etc/certs/ca.crt"},{verify,"verify_peer"},{fail_if_no_peer_cert,true}]) for reason {options,^M                                                                                                                                                                         {verify,^M                                                                                                                                                                       "verify_peer"}} (unknown POSIX error)^M
``

After checked out the verify options in emqx.conf, I find that the datatype of verify option is atom in emqx.conf while it is string in mgmt. Does it cause the problem emqx failed to start mgmt plugin with https listener?
```
{mapping, "management.listener.https.verify", "emqx_management.listeners", [
  {datatype, string}
]}.
```
```
{mapping, "listener.ssl.$name.verify", "emqx.listeners", [
 {datatype, atom}
]}.
```

**Comment by Gilbert-Wong** [comment]
Which emqx version do you use? And which os do you run emqx on?

**Comment by RicardoHPeng** [comment]
Sry, forgot to mention that. Version: 3.0.0, OS: Ubuntu 14.04, Trusty Tahr.
Seems changing the datatype in releases/3.0/schema/emqx_management.schema can fix the problem.

**Comment by Gilbert-Wong** [comment]
Did you place the cert file correctly on your specified place?

**Comment by RicardoHPeng** [comment]
Yes, I think so. And mgmt with https works after I modified the datatype 'verify'  to atom in emqx_management.schema. 
Btw, here is the mgmt.conf.
```
##--------------------------------------------------------------------
## EMQ X Management Plugin
##--------------------------------------------------------------------

## Max Row Limit
management.max_row_limit = 10000

##--------------------------------------------------------------------
## HTTP Listener

management.listener.http = 8080
management.listener.http.acceptors = 2
management.listener.http.max_clients = 512
management.listener.http.backlog = 512
management.listener.http.send_timeout = 15s
management.listener.http.send_timeout_close = on

##--------------------------------------------------------------------
## HTTPS Listener

 management.listener.https = 8081
 management.listener.https.acceptors = 2
 management.listener.https.max_clients = 512
 management.listener.https.backlog = 512
 management.listener.https.send_timeout = 15s
 management.listener.https.send_timeout_close = on
 management.listener.https.certfile = etc/certs/client.crt
 management.listener.https.keyfile = etc/certs/client.key
 management.listener.https.cacertfile = etc/certs/ca.crt
 management.listener.https.verify = verify_peer
 management.listener.https.fail_if_no_peer_cert = true
```
 And I put the cert files etc/certs dir.
```
ls -l etc/certs/
total 48
drwxr-xr-x 2 root root 4096 Jan 21 14:26 ./
drwxr-xr-x 5 root root 4096 Jan 21 17:17 ../
-rw-r--r-- 1 root root 1021 Dec 23 08:04 cacert.pem
-rw-r--r-- 1 root root 8015 Jan 21 14:26 ca.crt
-rw-r--r-- 1 root root 1070 Dec 23 08:04 cert.pem
-rw-r--r-- 1 root root 1070 Dec 23 08:04 client-cert.pem
-rw-r--r-- 1 root root 2394 Jan 21 14:26 client.crt
-rw-r--r-- 1 root root 1679 Jan 21 14:26 client.key
-rw-r--r-- 1 root root 1675 Dec 23 08:04 client-key.pem
-rw-r--r-- 1 root root 1675 Dec 23 08:04 key.pem
-rw-r--r-- 1 root root   38 Dec 23 08:04 README
```

**Comment by HJianBo** [comment]
Hi, @RicardoHPeng Sorry too late reply there. I have fixed the `verify` option datatype error: https://github.com/emqx/emqx-management/commit/8ad80dbec7bddbe4e53802ac71c9496fe244815e, and it will be released in the next version v3.1.1 after two weeks.

Thanks for your feedback!
